Dfe.SchoolAccount.Web/Services/Content/ContentfulSignpostingPageContentFetcher.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace Dfe.SchoolAccount.Web.Services.Content;
+
+using System.Threading.Tasks;
+using Contentful.Core;
+using Contentful.Core.Models;
+using Contentful.Core.Search;
+using Dfe.SchoolAccount.Web.Constants;
+using Dfe.SchoolAccount.Web.Models.Content;
+using Dfe.SchoolAccount.Web.Services.ContentTransformers;
+using Microsoft.AspNetCore.Connections.Features;
+
+/// <summary>
+/// A service which fetches signposting page content from a Contentful CDA.
+/// </summary>
+public sealed class ContentfulSignpostingPageContentFetcher : ISignpostingPageContentFetcher
+{
+    private readonly IContentfulClient contentfulClient;
+    private readonly IContentModelTransformer contentModelTransformer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ContentfulSignpostingPageContentFetcher"/> class.
+    /// </summary>
+    /// <param name="contentfulClient">The contentful client.</param>
+    /// <param name="contentModelTransformer">A service for transforming content models.</param>
+    /// <exception cref="ArgumentNullException">
+    /// If <paramref name="contentfulClient"/> or <paramref name="contentModelTransformer"/> is <c>null</c>.
+    /// </exception>
+    public ContentfulSignpostingPageContentFetcher(
+        IContentfulClient contentfulClient,
+        IContentModelTransformer contentModelTransformer)
+    {
+        if (contentfulClient == null) {
+            throw new ArgumentNullException(nameof(contentfulClient));
+        }
+        if (contentModelTransformer == null) {
+            throw new ArgumentNullException(nameof(contentModelTransformer));
+        }
+
+        this.contentfulClient = contentfulClient;
+        this.contentModelTransformer = contentModelTransformer;
+    }
+
+    /// <inheritdoc/>
+    public async Task<SignpostingPageContent?> FetchSignpostingPageContentAsync(string slug)
+    {
+        if (slug == null) {
```

Comment:
Existing DfE projects are doing it the more verbose approach and this would introduce a dependency. I've made a note of this and am happy to open a discussion but in general the fewer dependencies the better.